### DISCUSSION TITLE
MacOS Catalina 10.15.7 System.TypeInitializationException (DLL NotFound)

### DISCUSSION THREAD

**Discussion by grzesl** [dlemstra/Magick.NET#756] [Help]
Hello,

Im using Visual Studio for Mac and I download from NuGet 2 packages:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/3374092/95688546-73171c00-0c0b-11eb-9efe-8af037e7bfc1.png">

When I run line of code:

> using (MagickImage image = new MagickImage("tool.png"))

Im getting error:
<img width="570" alt="image" src="https://user-images.githubusercontent.com/3374092/95688327-b96b7b80-0c09-11eb-8c07-dc78e845ca36.png">

Should I initialize library before using it ore something?
Maybe my version of system is more restricted about using not signed library?

Have you got any idea what's I'm doing wrong?

**Comment by dlemstra** [comment]
I suspect that the library is not copied from the Native folder to the output? No idea why that did not happen though.

**Comment by grzesl** [comment]
Hello, 
Thanks for quick respond.

I have copy the native library from packages to Debug folder and run code with no success. I try to get some information from inside library and I have got:

<img width="768" alt="image" src="https://user-images.githubusercontent.com/3374092/95758405-ba9fb580-0ca8-11eb-9aa0-9b105f7e31e7.png">

Any idea?

**Reply by dlemstra** [reply]
Strange that you don't see the inner exception. Maybe the library cannot be loaded due to dependency issues? What is the output of `ld` and `ldd` for the native library?

**Comment by grzesl** [comment]
This is the output (correct me if You expected something else ):

(ldd)
> % otool -L Magick.Native-Q16-x64.dll.dylib 
> Magick.Native-Q16-x64.dll.dylib:
> 	@rpath/libMagick.Native-Q16-x64.dll.dylib (compatibility version 0.0.0, current version 0.0.0)
> 	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
> 	/usr/lib/libresolv.9.dylib (compatibility version 1.0.0, current version 1.0.0)
> 	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 902.1.0)
> 	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)

(ld)
> % ld Magick.Native-Q16-x64.dll.dylib 
> ld: warning: platform not specified
> ld: warning: -arch not specified
> ld: warning: No platform min-version specified on command line
> ld: warning: ignoring file Magick.Native-Q16-x64.dll.dylib, building for -unknown but attempting to link with file built for macOS-x86_64
> Undefined symbols for architecture unknown:
>   "_main", referenced from:
>      implicit entry/start for main executable
> ld: symbol(s) not found for architecture unknown

**Reply by dlemstra** [reply]
I don't have a mac so I don't know what to expect 😁The output looks okay to me.

**Comment by grzesl** [comment]
Hi, I finally get the library  running. In output of otool I found that library has name:
**lib**Magick.Native-Q16-x64.dll.dylib 
so I change native library file name from  
Magick.Native-Q16-x64.dll.dylib  -> **lib**Magick.Native-Q16-x64.dll.dylib 
and copy it to /usr/local/lib

Now library works. 
I don't know exactly what is current path in my project (Debug folder does not worked).

I think there is some native library name problem in the library.
